cirq-google/cirq_google/engine/stream_manager.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2023 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict
+import asyncio
+
+from google.api_core.exceptions import GoogleAPICallError
+
+from cirq_google.cloud import quantum
+
+
+class ResponseDemux:
+    """A event demultiplexer for QuantumRunStreamResponses, as part of the async reactor pattern."""
```

Comment:
Can you include more detail about the usage and expected behavior of this class? I had understood that demultiplexing would fan out the response to multiple subscribers, but it doesn't look like that's done here so I think I'm misunderstanding the intent :-).